PWG/EMCAL/EMCALtasks/AliEmcalCorrectionClusterHadronicCorrection.cxx

```diff
@@ -634,6 +637,19 @@ void AliEmcalCorrectionClusterHadronicCorrection::DoMatchedTracksLoop(Int_t iclu
     else {
       etaCut = GetEtaSigma(mombin);
     }
+    //Do momentum dependent track matching. Values taken from the PCM analyses see:
+    //https://alice-notes.web.cern.ch/node/411  Fig. 46   for mom<3GeV these cuts are wider than the standard cuts
+    //these values were extracted in the 2012 pp8 TeV MonteCarlo and apparently showed robustness throughout different periods
+    if (fDoMomDepMatching){
+      TF1* fFuncPtDepEta = new TF1("func", "[1] + 1 / pow(x + pow(1 / ([0] - [1]), 1 / [2]), [2])");
+      fFuncPtDepEta->SetParameters(0.04, 0.010, 2.5);
+      TF1* fFuncPtDepPhi = new TF1("func", "[1] + 1 / pow(x + pow(1 / ([0] - [1]), 1 / [2]), [2])");
+      fFuncPtDepPhi->SetParameters(0.09, 0.015, 2.);
```

Comment:
Please move the definition of these functions above the for loop so they don't have to be defined every loop. To avoid `new`, here is what I suggest:

```c++
// Just define the fit objects here. They are not necessary if pt dependent matching isn't used, but it also doesn't cost much to do it once.
TF1 funcPtDepEta("func", "[1] + 1 / pow(x + pow(1 / ([0] - [1]), 1 / [2]), [2])");
...

for (...)
{
  if (fDoMomDepMatching) {
    phiCutLo = -funcPtDepPhi->Eval(mom);
    ...
  }
  ...
}
```

Reply:
I added your suggestion